pkg/util/utils.go

```diff
@@ -723,3 +726,316 @@ func SocketPath() (string, error) {
+		cmd.Args = append(cmd.Args, parentFlags...)
+		cmd.Args = append(cmd.Args, loadCommand...)
+	} else {
+		cmd = exec.Command("sudo", podman)
+		cmd.Args = append(cmd.Args, parentFlags...)
+		cmd.Args = append(cmd.Args, loadCommand...)
+	}
+	cmd.Env = os.Environ()
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	logrus.Debug("Executing load command")
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TransferRootless creates new podman processes using exec.Command and su/machinectl, transferring images between the given source and destination users
+func TransferRootful(source ImageScpOptions, dest ImageScpOptions, podman string, parentFlags []string) error {
+	saveCommand := []string{"save", "--output", source.File, source.Image}
+	loadCommand := []string{"load", "--input", dest.File}
+	// if executing using sudo or transferring between two users, the TransferRootless approach will not work, default to using machinectl or su as necessary.
+	// the approach using sudo is preferable and more straightforward. There is no reason for using sudo in these situations
+	// since the feature is meant to transfer from root to rootless an vice versa without explicit sudo evocaiton.
+	var uSave *user.User
+	var uLoad *user.User
+	source.User = strings.Split(source.User, ":")[0] // split in case provided with uid:gid
+	dest.User = strings.Split(dest.User, ":")[0]
+	_, err := strconv.Atoi(source.User)
+	if err != nil {
+		uSave, err = user.Lookup(source.User)
+		if err != nil {
+			return err
+		}
+	} else {
+		uSave, err = user.LookupId(source.User)
+		if err != nil {
+			return err
+		}
+	}
+	if dest.User != "" { // if we are given a destination user, check that first
+		_, err := strconv.Atoi(dest.User)
+		if err != nil {
+			uLoad, err = user.Lookup(dest.User)
+			if err != nil {
+				return err
+			}
+		} else {
+			uLoad, err = user.LookupId(dest.User)
+			if err != nil {
+				return err
+			}
+		}
+	} else if uSave.Name != "root" { // else if we have no destination user, and source is not root that means we should be root
+		uLoad, err = user.LookupId("0")
+		if err != nil {
+			return err
+		}
+	} else { // else if we have no dest user, and source user IS root, we want to be the default user.
+		uString := os.Getenv("SUDO_USER")
+		if uString == "" {
+			uLoad, err = user.LookupId("1000")
+			if err != nil {
+				return err
+			}
+		} else {
+			uLoad, err = user.Lookup(uString)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	var cmd *exec.Cmd
+	machinectl, err := exec.LookPath("machinectl")
+	if err != nil {
+		logrus.Warn("defaulting to su since machinectl is not available, su will fail if no user session is available")
+		cmd = exec.Command("su", "-l", uSave.Username, "--command")
+		fullCommand := podman
+		for _, val := range parentFlags {
+			fullCommand += (" " + val)
+		}
+		for _, val := range saveCommand {
+			fullCommand += (" " + val)
+		}
+		cmd.Args = append(cmd.Args, fullCommand)
```

Comment:
This same code seems to be copied and pasted multiple times, can you just create a function and pass in the command?